pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -115,6 +115,16 @@ func ConvertService(svc coreV1.Service, domainSuffix string, clusterID string) *
 		},
 	}
 
+	if svc.Spec.Type == coreV1.ServiceTypeNodePort {
+		// store the service port to node port mappings
+		portMap := make(map[uint32]uint32)
+		for _, p := range svc.Spec.Ports {
```

Comment:
IC, the gateway service has many ports, though we only need one to do redirect, we are not sure which one it is.

Reply:
yep